winnowing.py

```diff
@@ -3,122 +3,123 @@
+        self.HL1 = self.hashList(self.kGrams1)  #hash list derived from k-grams list
+        self.HL2 = self.hashList(self.kGrams2)
+        self.fpList1 = self.fingerprints(self.HL1)
+        self.fpList2 = self.fingerprints(self.HL2)
 
-#function to form k-grams out of the cleaned up text
-def kgrams(text, k = 25):
-    tokenList = list(text)
-    n = len(tokenList)
-    kgrams = []
-    for i in range(n - k + 1):
-        kgram = ''.join(tokenList[i : i + k])
-        hv = hash(kgram)
-        kgrams.append((kgram, hv, i, i + k))  #k-gram, its hash value, starting and ending positions are stored
-        #these help in marking the plagiarized content in the original code.
-    return kgrams
-
-#function that returns the index at which minimum value of a given list (window) is located
-def minIndex(arr):
-    minI = 0
-    minV = arr[0]
-    n = len(arr)
-    for i in range(n):
-        if arr[i] < minV:
-            minV = arr[i]
-            minI = i
-    return minI
+    def plagiarismCheck(self):
+        file1 = self.file1
+        file2 = self.file2
+        f1 = open(file1, "r")
+        start = []   #to store the start values corresponding to matching fingerprints
+        end = []   #to store end values
+        code = f1.read()  #original code
+        newCode = ""   #code with marked plagiarized content
+        points = []
+        for i in self.fpList1:
+            for j in self.fpList2:
+                if i == j:   #fingerprints match
+                    flag = 0
+                    match = self.HL1.index(i)   #index of matching fingerprints in hash list, k-grams list
+                    newStart = self.kGrams1[match][2]   #start position of matched k-gram in cleaned up code
+                    newEnd = self.kGrams1[match][3]   #end position
+                    for k in self.token1:
+                        if k[2] == newStart:   #linking positions in cleaned up code to original code
+                            startx = k[1]
+                            flag = 1
+                        if k[2] == newEnd:
+                            endx = k[1]
+                    if flag == 1:
+                        points.append([startx, endx])
+        points.sort(key = lambda x: x[0])
+        points = points[1:]
+        mergedPoints = []
+        mergedPoints.append(points[0])
+        for i in range(1, len(points)):
+            last = mergedPoints[-1]
+            if points[i][0] >= last[0] and points[i][0] <= last[1]: #merging overlapping regions
+                if points[i][1] > last[1]:
+                    mergedPoints = mergedPoints[:-1]
+                    mergedPoints.append([last[0], points[i][1]])
+                else:
+                    pass
+            else:
+                mergedPoints.append(points[i])
+        newCode = code[: mergedPoints[0][0]]
+        plagCount = 0
+        for i in range(len(mergedPoints)):
+            if mergedPoints[i][1] > mergedPoints[i][0]:
```

Comment:
This syntax is C styled. Also, you don't need to indexing python can directly get the elements from a list. No need of doing range(len(mergedPoints)).

Reply:
solved sir.